Shared/Realm.Shared/native/SynchronizationContextEventLoopSignal.cs

```diff
@@ -0,0 +1,117 @@
+﻿////////////////////////////////////////////////////////////////////////////
+//
+// Copyright 2016 Realm Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+////////////////////////////////////////////////////////////////////////////
+
+using System;
+using System.Diagnostics;
+using System.Diagnostics.CodeAnalysis;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+using System.Threading;
+
+namespace Realms
+{
+    [SuppressMessage("StyleCop.CSharp.NamingRules", "SA1300:ElementMustBeginWithUpperCaseLetter")]
+    internal static class SynchronizationContextEventLoopSignal
+    {
+        private delegate IntPtr get_eventloop();
+
+        private delegate void post_on_event_loop(IntPtr eventloop, EventLoopPostHandler callback, IntPtr user_data);
+
+        private delegate void release_eventloop(IntPtr eventloop);
+
+        private delegate void EventLoopPostHandler(IntPtr user_data);
+
+        [DllImport(InteropConfig.DLL_NAME, EntryPoint = "realm_install_eventloop_callbacks", CallingConvention = CallingConvention.Cdecl)]
+        private static extern void install_eventloop_callbacks(get_eventloop get, post_on_event_loop post, release_eventloop release);
+
+        private class EventLoop
+        {
+            private SynchronizationContext _context;
+
+            private volatile bool _isReleased = false;
+
+            internal EventLoop(SynchronizationContext context)
+            {
+                Debug.Assert(context != null, "The SynchronizationContext EventLoopSignal implementation always needs a SynchronizationContext");
+                _context = context; 
+            }
+
+            internal void Post(EventLoopPostHandler callback, IntPtr user_data)
+            {
+                _context.Post(_ =>
```

Comment:
Now that the `_context` can be null, perhaps use null-propagation just to be extra safe? :D